### DISCUSSION TITLE
Cannot start picom with "glx" backend

### DISCUSSION THREAD

**Discussion by dnquinn** [termux/termux-packages#20284] [Q&A]
Hello, has anyone encountered this problem before? For some reason, I can't get picom to work with the OpenGL driver on termux xfce (non-proot).

When I run `picom --diagnostic`, I get the following error:

```
### Backend: glx

[ 05/21/24 19:11:43.702 glx_init ERROR ] Failed to get GLX context. Cannot initialize this backend

### Backend: egl

libEGL warning: DRI3: Screen seems not DRI3 capable
[ 05/21/24 19:11:43.844 egl_init ERROR ] EGL version too old, need at least 1.5. Cannot initialize this backend
```

However, the drivers seem to work and `glxinfo` outputs the following information:

```
Device: virgl (ANGLE (Qualcomm, Adreno (TM) 650, OpenGL ES 3.2 V@05...) (0x1010)
OpenGL vendor string: Mesa
OpenGL renderer string: virgl (ANGLE (Qualcomm, Adreno (TM) 650, OpenGL ES 3.2 V@05...)
OpenGL version string: 2.1 Mesa 24.0.7
OpenGL shading language version string: 1.20
OpenGL extensions:
OpenGL ES profile version string:
OpenGL ES 3.1 Mesa 24.0.7
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.10

```

If I use xrender as the backend, picom works, but the FPS drops significantly, and proper blur effects cannot be achieved.

I've tried reinstalling the mesa and virgl drivers, changing launch parameters, and editing the config, but unfortunately, nothing has worked.

Would be very appreciated for any help :)

**Comment by jorbig** [comment]
Any updates?

**Comment by funsafe-ptr** [comment]
I'm seeing the same error, but when I run `apt info libglvnd`, it says the version is `1.7.0`? I'm not sure.
